shim/src/handle_system_kvm_create_vm.c

```diff
@@ -25,17 +25,36 @@
  */
 
 #include <debug.h>
+#include <g_hndl.h>
+#include <mv_hypercall.h>
+#include <platform.h>
 #include <types.h>
 
+/* Remove me */
+static uint16_t
+mv_vm_op_create_vm(uint64_t const g_hndl)
+{
+    (void)g_hndl;
+    return 1;
+}
+
 /**
  * <!-- description -->
- *   @brief Handles the execution of kvm_check_extension.
+ *   @brief Handles the execution of kvm_create_vm.
  *
  * <!-- inputs/outputs -->
  *   @return SHIM_SUCCESS on success, SHIM_FAILURE on failure.
  */
 int64_t
-handle_system_kvm_create_vm(void)
+handle_system_kvm_create_vm(uint16_t *const vmid)
```

Comment:
you should be passing in a pointer to the shim_vm_t, not the vmid itself. Then set the vmid using vm->vmid = .....

Reply:
Also, change the code to this:
```
    platform_expects(MV_INVALID_HANDLE != g_hndl);
    platform_expects(NULL != vm);

   vm->vmid = mv_vm_op_create_vm(g_hndl);
    if (MV_INVALID_ID == vm->vmid ) {
        bferror("mhandle_system_kvm_create_vm:: mv_vm_op_create_vm failed");
        return SHIM_FAILURE;
    }
```

The reason is that the error is associated with the call to mv_xxx, so it should be grouped with this call.

Reply:
Also, the string in the error message is corrupt